js/gestionPresupuesto.js

```diff
@@ -1,25 +1,124 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
 
-// TODO: Variable global
+let presupuesto = 0;
+let gastos = [];
+let idGasto = 0;
 
+function actualizarPresupuesto(cantidad) {
+  if (cantidad >= 0) {
+    presupuesto = cantidad;
+    return presupuesto;
+  } else {
+    console.log("Se ha introducido una cantidad negativa.");
+    return -1;
+  };
 
-function actualizarPresupuesto() {
-    // TODO
-}
+};
 
 function mostrarPresupuesto() {
-    // TODO
-}
+  return `Tu presupuesto actual es de ${presupuesto} €`;
+};
+
+function CrearGasto(descripcion, gasto, fecha = Date.now(), ...etiquetas) {
+  this.descripcion = descripcion;
+
+  if (gasto >= 0) {
+    this.valor = gasto;
+  } else {
+    this.valor = 0;
+  }
+
+  if (isNaN(Date.parse(fecha))) {
+    this.fecha = Date.now();
+  } else {
+    this.fecha = Date.parse(fecha);
+  }
+
+  if (etiquetas.length == 0) {
+    this.etiquetas = [];
+  } else {
+    this.etiquetas = etiquetas
```

Comment:
En lugar de esto, puedes utilizar la función `this.anyadirEtiquetas` para evitar crear un gasto con etiquetas duplicadas.